table/tables/index.go

```diff
@@ -394,7 +397,7 @@ func (c *index) Create(sctx sessionctx.Context, rm kv.RetrieverMutator, indexedV
 		idxVal = make([]byte, 0)
 		if distinct {
 			if h.IsInt() {
-				idxVal = EncodeHandleInUniqueIndexValue(h.IntValue())
+				idxVal = EncodeHandleInUniqueIndexValue(h)
```

Comment:
If we add `untouched` parameter to `EncodeHandleInUniqueIndexValue`, we can call it out of the `if h.IsInt` block